crates/driver/src/tests/setup/solver.rs

```diff
@@ -1,63 +1,235 @@
-    pub solve: Vec<Solve>,
-    pub name: String,
-    pub absolute_slippage: String,
-    pub relative_slippage: String,
-    pub address: String,
-    pub private_key: String,
-}
+pub const NAME: &str = "test-solver";
 
-#[derive(Debug, Clone)]
-pub struct Solve {
-    pub req: serde_json::Value,
-    pub res: serde_json::Value,
+pub struct Solver {
+    pub addr: SocketAddr,
 }
 
-/// A solver that was started as part of the setup.
 #[derive(Debug)]
-pub struct Solver {
-    pub config: Config,
-    pub addr: SocketAddr,
+pub struct Config<'a> {
+    pub blockchain: &'a Blockchain,
+    pub solutions: &'a [blockchain::Solution],
+    pub trusted: &'a HashSet<&'static str>,
+    pub quotes: &'a [super::blockchain::Quote],
+    pub deadline: chrono::DateTime<chrono::Utc>,
+    pub now: infra::time::Now,
+    /// Is this a test for the /quote endpoint?
+    pub quote: bool,
 }
 
-/// Set up an HTTP server exposing a solver API and acting as a solver mock.
-pub async fn setup(config: Config) -> Solver {
-    let state = Arc::new(Mutex::new(config.solve.clone()));
-    let app = axum::Router::new()
+impl Solver {
+    /// Set up an HTTP server exposing a solver API and acting as a solver mock.
+    pub async fn new(config: Config<'_>) -> Self {
+        let mut solutions_json = Vec::new();
+        let mut orders_json = Vec::new();
+        for quote in config.quotes {
+            // ETH orders get unwrapped into WETH by the driver before being passed to the
+            // solver.
+            let sell_token = if quote.order.sell_token == "ETH" {
+                "WETH"
+            } else {
+                quote.order.sell_token
+            };
+            let buy_token = if quote.order.buy_token == "ETH" {
+                "WETH"
+            } else {
+                quote.order.buy_token
+            };
+            orders_json.push(json!({
+                "uid": if config.quote { Default::default() } else { quote.order_uid(config.blockchain, config.now) },
+                "sellToken": hex_address(config.blockchain.get_token(sell_token)),
+                "buyToken": hex_address(config.blockchain.get_token(buy_token)),
+                "sellAmount": match quote.order.side {
+                    order::Side::Buy if config.quote => "6277101735386680763835789423207666416102355444464034512896".to_owned(),
+                    _ => (quote.sell_amount() - quote.order.surplus_fee()).to_string(),
+                },
+                "buyAmount": match quote.order.side {
+                    order::Side::Sell if config.quote => "1".to_owned(),
+                    _ => quote.buy_amount().to_string(),
+                },
+                "feeAmount": quote.order.user_fee.to_string(),
+                "kind": match quote.order.side {
+                    order::Side::Sell => "sell",
+                    order::Side::Buy => "buy",
+                },
+                "partiallyFillable": matches!(quote.order.partial, order::Partial::Yes { .. }),
```

Comment:
nit: I believe in case of `config.quote` this needs to be `false`. A quote for a partial fill is pretty much useless for our purposes. Plus you already correctly set `class: market` in this case which is incompatible with `partiallFillable: true`.